tests/posterior_nn_test.py

```diff
@@ -315,6 +317,60 @@ def test_batched_score_sample_with_different_x(
         )
 
 
+@pytest.mark.slow
+@pytest.mark.parametrize("vector_field_method", [NPSE, FMPE])
+@pytest.mark.parametrize("x_o_batch_dim", (0, 1, 2))
+@pytest.mark.parametrize("sampling_method", ["sde", "ode"])
+@pytest.mark.parametrize(
+    "sample_shape",
+    (
+        (5,),  # less than num_chains
+        (4, 2),  # 2D batch
+    ),
+)
+def test_batched_vector_field_sample_with_multidim_x(
```

Comment:
can you please explain what this test is doing different? Is this new test a more general one so that we can remove the test above? 
if not, why do we have to test NPSE as well?